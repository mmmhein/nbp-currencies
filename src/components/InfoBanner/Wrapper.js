import styled from 'styled-components';

import { colorTurquoise, colorWhite, fontMedium, spacingSmall, colorDarkGrey } from '../../styles/designTokens';

const Wrapper = styled.section`
  display: flex;
  justify-content: center;
  padding: ${spacingSmall} 0;
  background-color: ${({ grey }) => (grey ? colorDarkGrey : colorTurquoise)};
  color: ${colorWhite};
  font-size: ${fontMedium};
  text-align: center;
`;

export default Wrapper;
