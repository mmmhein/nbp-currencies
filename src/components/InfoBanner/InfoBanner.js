import React from 'react';
import PropTypes from 'prop-types';

import Content from '../Content';
import Wrapper from './Wrapper';

function InfoBanner({ grey, children }) {
  return (
    <Wrapper grey={grey}>
      <Content as="div">{children}</Content>
    </Wrapper>
  );
}

InfoBanner.propTypes = {
  red: PropTypes.bool,
  children: PropTypes.node
};

export default InfoBanner;
