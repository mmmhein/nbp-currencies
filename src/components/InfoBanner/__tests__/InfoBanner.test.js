import React from 'react';
import { create } from 'react-test-renderer';

import InfoBanner from '../InfoBanner';

describe('COMPONENT - InfoBanner', () => {
  it('renders correctly', () => {
    const component = create(<InfoBanner grey />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with grey background', () => {
    const component = create(<InfoBanner grey />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
