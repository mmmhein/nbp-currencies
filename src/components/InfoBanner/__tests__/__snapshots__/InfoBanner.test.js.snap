// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - InfoBanner renders correctly 1`] = `
.c1 {
  max-width: 100rem;
  padding: 0 2rem;
  margin: 0 auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  background-color: #666666;
  color: #ffffff;
  font-size: 1.8rem;
  text-align: center;
}

<section
  className="c0"
>
  <div
    className="c1"
  />
</section>
`;

exports[`COMPONENT - InfoBanner renders correctly with grey background 1`] = `
.c1 {
  max-width: 100rem;
  padding: 0 2rem;
  margin: 0 auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  background-color: #666666;
  color: #ffffff;
  font-size: 1.8rem;
  text-align: center;
}

<section
  className="c0"
>
  <div
    className="c1"
  />
</section>
`;
